src/signal_scope/plotwidget.cpp

```diff
@@ -240,6 +247,47 @@ double PlotWidget::timeWindow() const
   return this->d_plot->timeWindow();
 }
 
+bool PlotWidget::getScale()
+{
+  float tmpRange[2];
+  bool update = false;
+
+  for (int ii=0; ii<mSignalListWidget->count(); ii++){
+    QListWidgetItem* signalItem = mSignalListWidget->item(ii);
+    SignalHandler* signalHandler = signalItem->data(Qt::UserRole).value<SignalHandler*>();
+    SignalData* signalData = signalHandler->signalData();
+
+    size_t nn = signalData->size();
+    if (nn){
+      tmpRange[0] = signalData->value(0).y();
+      tmpRange[1] = signalData->value(0).y();
+
+      for (size_t jj=1; jj<signalData->size(); jj++){
+        if (signalData->value(jj).y() > tmpRange[1])
+          tmpRange[1] = signalData->value(jj).y();
+        else if (signalData->value(jj).y() < tmpRange[0])
+          tmpRange[0] = signalData->value(jj).y();
+      }      
```

Comment:
i'm not sure why this function needs to compute the range?  The signalData object has a function called computeBounds() which returns the y range.  can you use that?